code/game/objects/items/tools/welder.dm

```diff
@@ -153,19 +153,34 @@
 		return
 	remove_fuel(0.5)
 
-/obj/item/weldingtool/attack(mob/living/carbon/M, mob/living/carbon/user)
-	// For lighting other people's cigarettes.
-	var/obj/item/clothing/mask/cigarette/cig = M?.wear_mask
-	if(!istype(cig) || user.zone_selected != "mouth" || !tool_enabled)
+/obj/item/weldingtool/attack(mob/living/target, mob/living/user)
+	if(!cigarette_lighter_act(user, target))
 		return ..()
 
-	if(M == user)
-		cig.attackby(src, user)
-		return
+/obj/item/weldingtool/cigarette_lighter_act(mob/living/user, mob/living/target, obj/item/direct_attackby_item)
+	var/obj/item/clothing/mask/cigarette/cig = ..()
+	if(!cig)
+		if(isnull(cig))
+			return
+		else
+			return TRUE
```

Comment:
Why not do
```suggestion
	if(!cig)
		return isnull(cig)

```

Reply:
This returns the opposite of the desired behaviour (you attack someone with a lit cig, and don't attack someone with no cig).

`return !isnull(cig)` does work, however.

Reply:
All have been changed.